lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,227 @@
+              // Add a map view to the widget tree and set a controller.
+              child: ArcGISMapView(
+                controllerProvider: () => _mapViewController,
+                onMapViewReady: onMapViewReady,
+              ),
+            ),
+            // Add a Settings button to the widget tree.
+            ElevatedButton(
+              // Show the settings dialog when the button is pressed.
+              onPressed: _ready
+                  ? () => showDialog(
+                        context: context,
+                        builder: (context) => showSettings(context),
+                      )
+                  : null,
+              child: const Text(
+                'Settings',
+                style: TextStyle(
+                  color: Colors.deepPurple,
+                ),
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a portal item.
+    final portal = Portal.arcGISOnline();
+    final portalItem = PortalItem.withPortalAndItemId(
+        portal: portal, itemId: '3953413f3bd34e53a42bf70f2937a408');
+    // Load the portal item.
+    await portalItem.load();
+
+    // Create a map from the portal item and load it.
+    _map = ArcGISMap.withItem(portalItem);
+    await _map.load();
+
+    // Get the operational layers from the map.
+    for (var layer in _map.operationalLayers) {
+      _allFeatureLayers.add(layer.name);
+    }
+    _mapViewController.arcGISMap = _map;
+
+    // See if the map is ready.
+    setState(() => _ready = true);
+  }
+
+  StatefulBuilder showSettings(BuildContext newContext) {
+    // Show the settings dialog.
+    return StatefulBuilder(builder: (newContext, setNewState) {
+      return Dialog(
+        child: Padding(
+          padding: const EdgeInsets.all(8),
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: [
+              Center(
+                child: Text('Settings',
+                    style: Theme.of(context).textTheme.headlineMedium),
+              ),
+              // Add a dropdown button for setting a new reference scale.
+              DropdownButton(
+                alignment: Alignment.center,
+                hint: const Text(
+                  'Select a New Reference Scale',
+                  style: TextStyle(
+                    color: Colors.deepPurple,
+                  ),
+                ),
+                // Set the selected scale
+                value: _selectedScale,
+                icon: const Icon(
+                  Icons.arrow_drop_down,
+                  color: Colors.deepPurple,
+                ),
+                elevation: 16,
+                style: const TextStyle(color: Colors.deepPurple),
+                // Set the onChanged callback to update the selected scale.
+                onChanged: (newScale) {
+                  updateScale(newScale!.toDouble());
+                },
```

Comment:
I got rid of the updateScale() function but `newScale` is num so we would need to be casting

Reply:
Maybe it needs casting, like `newScale as double`. But it should be smart enough to know it's a double.

Actually, I think I know what it is -- _referenceScaleList uses `double`, but _selectedScale uses `int`. That confuses the compiler and the best it can come up with is `num`. I bet if _selectedScale was a double, the compiler would figure out that newScale is supposed to be a double.

And actually it might make sense to get rid of _selectedScale, and just use _scale in its place. Yeah _selectedScale doesn't really do anything, and _scale is actually doing the work.

Reply:
Yeap you are right! I removed _selectedScale and it solved the problem with casting, great suggestion Paul!